patches/minecraft/net/minecraft/world/gen/ChunkGeneratorOverworld.java.patch

```diff
@@ -100,3 +109,39 @@
          BlockFalling.field_149832_M = false;
      }
  
+@@ -475,7 +508,8 @@
+     public List<Biome.SpawnListEntry> func_177458_a(EnumCreatureType p_177458_1_, BlockPos p_177458_2_)
+     {
+         Biome biome = this.field_185995_n.func_180494_b(p_177458_2_);
+-
++        net.minecraftforge.event.terraingen.ChunkGeneratorEvent.GetPossibleCreatures event=new net.minecraftforge.event.terraingen.ChunkGeneratorEvent.GetPossibleCreatures(this, field_185995_n, p_177458_1_, p_177458_2_);
++        if(net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event)) return event.getPossibleCreatures();
+         if (this.field_185996_o)
+         {
+             if (p_177458_1_ == EnumCreatureType.MONSTER && this.field_186007_z.func_175798_a(p_177458_2_))
+@@ -494,6 +528,8 @@
+ 
+     public boolean func_193414_a(World p_193414_1_, String p_193414_2_, BlockPos p_193414_3_)
+     {
++        net.minecraftforge.event.terraingen.ChunkGeneratorEvent.StructureCheck event = new net.minecraftforge.event.terraingen.ChunkGeneratorEvent.StructureCheck(this, field_185995_n, p_193414_2_, p_193414_3_);
++        if(net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event)) return event.isInStructure();
+         if (!this.field_185996_o)
+         {
+             return false;
+@@ -527,6 +563,8 @@
+     @Nullable
+     public BlockPos func_180513_a(World p_180513_1_, String p_180513_2_, BlockPos p_180513_3_, boolean p_180513_4_)
+     {
++        net.minecraftforge.event.terraingen.ChunkGeneratorEvent.FindStructure event = new net.minecraftforge.event.terraingen.ChunkGeneratorEvent.FindStructure(this, p_180513_1_, p_180513_2_, p_180513_3_, p_180513_4_);
++        if(net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event)) return event.getFoundPos();
+         if (!this.field_185996_o)
+         {
+             return null;
+@@ -559,6 +597,7 @@
+ 
+     public void func_180514_a(Chunk p_180514_1_, int p_180514_2_, int p_180514_3_)
+     {
++        net.minecraftforge.event.ForgeEventFactory.onChunkGenerate(this, this.field_185995_n, p_180514_2_, p_180514_3_, (ChunkPrimer) null);
```

Comment:
You don't need that cast.
Also, why is this the only one that has a bouncer method in `ForgeEventFactory`?
If anything, the ones above should have those, then you only need one line patches there.

Reply:
The cast is not really necessary, but wanted to keep it Vanilla style, also it doesn't hurt, does it?

I am not using a bouncer method here since this does not have a result. The other events can be cancelled to return a custom result, even if I would create a bouncer method for those, it would need require some way to tell the patched code if a) it was cancelled b)  what the result is. I could e.g. return null when it was not cancelled, but this would still require two lines of patched code, so I think it would be pointless. The rest of Forge's code seems to handle things like this similar.

I can of course remove  this bouncer method if you want

Reply:
> The cast is not really necessary, but wanted to keep it Vanilla style, also it doesn't hurt, does it?

It is not "vanilla style", it is "shitty decompiler style".
> I am not using a bouncer method here since this does not have a result.

But you _are_ using a bouncer method...

All the other event firing patches you added can be reduced to something like:
`if (someBouncerMethod(args)) return;` which is a one line patch.

Reply:
> I am not using a bouncer method here since this does not have a result.
This was of course meant the other way around:  I am using a bouncer method here since this does not have a result.

When firing the events I want:
- If they are not cancelled -> continue
- If they are cancelled -> return value from the event object (position, boolean, list ...)
This does not work with
>  `if (someBouncerMethod(args)) return;` which is a one line patch.

or am I missing something?